src/basho_bench_stats_writer_csv.erl

```diff
@@ -133,6 +172,53 @@ measurement_csv_file({Label, _Op}) ->
     ok = file:write(F, <<"elapsed, window, n, min, mean, median, 95th, 99th, 99_9th, max, errors\n">>),
     F.
 
+
+
+dump_run_statistics(RunStatsType) ->
+    Lines = stringify_stats(RunStatsType, erlang:get(run_metrics)),
+    TestDir = basho_bench:get_test_dir(),
+    FileName = filename:join([TestDir, "run_statistics." ++ atom_to_list(RunStatsType)]),
+    write_run_statistics(FileName, Lines).
+
+stringify_stats(_RunStatsType=json, RunMetrics) ->
+    [ jiffy:encode( {[{recordedMetrics, {RunMetrics}}]}, [pretty] ) ];
+stringify_stats(_RunStatsType=csv, RunMetrics) ->
+    %% Generate list of every known header
+    Headers = lists:foldl(fun(K, Acc) ->
+        {MetricsObj} = proplists:get_value(K, RunMetrics),
+        proplists:get_keys(MetricsObj) ++ Acc
+    end, [], proplists:get_keys(RunMetrics)),
+
+    %% Remove all duplicates and sort list of available headers, write to file
+    UniqueHeaders = lists:usort(Headers),
+    StringyfiedHeader =  io_lib:format("~s\n", 
+        [string:join( 
+           lists:map(fun atom_to_list/1, [op|UniqueHeaders]), ", ") ]),
+
+    %% Loop over each metric and build each line of the CSV file and write it out
+    Lines = lists:foldl(fun(Op, Lines) ->
+        {MetricsObj} = proplists:get_value(Op, RunMetrics),
+        CsvLine = lists:foldl(fun(K, Acc) ->
+            StrVal = io_lib:format("~w", [proplists:get_value(K, MetricsObj, "")]),
+            [StrVal|Acc]
+        end, [], UniqueHeaders),
+
+        [io_lib:format("~s\n", [string:join([atom_to_list(Op)|CsvLine],", ")])|Lines]
+    end, [], proplists:get_keys(RunMetrics)),
+    [StringyfiedHeader|Lines].
+
+write_run_statistics(FileName, Lines)->
+    {ok, GlobalMetricsFile} = file:open(
+        FileName,
+        [raw, binary, write]
+    ),
+    
+    %% Write lines to file
+    lists:foreach(fun(Line) ->
+        file:write(GlobalMetricsFile, Line)
```

Comment:
I would check the return value here like so: `ok = file:write(GlobalMetricsFile, Line)`.